fdbclient/NativeAPI.actor.cpp

```diff
@@ -1019,6 +1020,7 @@ void stopNetwork() {
 
 	g_network->stop();
 	closeTraceFile();
+	TLS::DestroyOpenSSLGlobalState();
```

Comment:
@mpilman has recently talked about adding a hook for when the network stops. I think what you have is fine as written, but I'm curious if you would have preferred to use the hook if it existed. If so, it could be extra motivation for him to add it.

Reply:
I have #2975 which contains this change. However we decided to go with `thread_local` instead. But having `Future<Void> INetwork::onShutdown()` could be useful in order places... Do you want me to make a PR for this? My plan was to wait until we have a better use-case...

Reply:
I added a `g_network->addStopCallback( []() { return; } );` in a parallel PR, that you're welcome to mutate to however you wish in the future.

Reply:
It actually turns out it was necessary to do the callback, as g_network->stop() doesn't wait for the network to actually stop, so I managed to re-introduce the same race condition I was trying to fix, and my testing happened to not reproduce it at the time.

But it is now, and I've reconfirmed that moving to the callback fixes things, so, master will be not broken again shortly.